client/allocrunner/network_hook.go

```diff
@@ -168,8 +169,14 @@ CREATE:
 
 			return fmt.Errorf("failed to configure networking for alloc: %v", err)
 		}
+		// A nil status indicates a netns already exists and is configured correctly.
+		// It should have been saved to the local state store.
 		if status == nil {
-			return nil // netns already existed and was correctly configured
+			stateStatus := h.networkStatus.NetworkStatus()
+			if stateStatus == nil {
+				return errors.New("network already configured but not found in state")
```

Comment:
how long ago was this added to state?  I'm wondering if this might be quite likely during upgrades (and cause a bunch of reschedules? is that what would happen?), or if it's been here for ages so probably safe.

if we have reason to err on the side of caution, this could be a log and return nil.